parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java

```diff
@@ -66,6 +66,9 @@ class ParquetWriter<T> implements FileAppender<T>, Closeable {
   private boolean closed;
   private ParquetFileWriter writer;
   private int rowGroupOrdinal;
+  private long currentRawBufferedSize = 0;
```

Comment:
So for my own clarity this is

currentRawBufferedSize = The amount of "buffer" in the currentRowGroup calculated as a diff in getBufferedSize()
toatalRawBufferedSIze = The amount of "buffer" in all RowGroups calculated as a diff in getBufferedSize()
totalRowGroupSize = The amount of "buffer" in all RowGroups calculated by just adding getBufferedSize() for each rowgroup together

Reply:
Is this right?

Reply:
each row group data has two different sizes: 
1. in memory before written (in_memory_size)
2. written as complete row group (written_size)

currentRawBufferedSize: "in_memory" of current unfinished row group 
toatalRawBufferedSIze: sum(in_memory for rg in finished_row_groups)
totalRowGroupSize: sum(written for rg in finished_row_groups)

we use (toatalRawBufferedSIze / totalRowGroupSize) ratio to estimate size change from in-memory to written state.

Reply:
how about 
`currentRawBufferedSize` -> `currentRowGroupMemoryBufferSize`
`totalRawBufferedSize` -> `cumulativeMemoryBufferSize`
`totalRowGroupSize` -> `cumulativeWrittenSize`